c/typedb_driver.i

```diff
@@ -132,11 +196,11 @@ static void transaction_callback_execute(size_t ID, Error* error) {
 %ignore transaction_on_close;
 %inline %{
 #include <atomic>
-void transaction_on_close_register(const Transaction* transaction, TransactionCallbackDirector* handler) {
+VoidPromise* transaction_on_close_register(const Transaction* transaction, TransactionCallbackDirector* handler) {
```

Comment:
`on_close` callback registration also returns a promise, since we're not guaranteed that the on_close callback is registered when this returns.